### PR TITLE
[TensorExpr] Port NNC lowerings to the new registry mechanism.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65555
* #65859
* #65852
* #65843
* #65554
* #65553
* #65552
* __->__ #65551
* #65550
* #65549

Previously we had a big switch on Op kind to decide how to lower a given
JIT operator to NNC. This PR changes this switch to a hash table lookup.

Why? This helps us with at least two things:
1) With this approach we can easily check if we know how to handle a
given node in advance - i.e. we can inspect the entire graph and tell
whether it's possible to compile it or not without actually trying to do
that and dying in the middle. This would allow us to, say, provide
user-friendly error messages in AOT workflow.
2) We can switch to use schema instead of op kind to determine correct
lowering. Unlike op schema, op kind might be ambigous (see e.g. #64963)
and using it instead of schema can lead to bugs.

Differential Revision: [D31148926](https://our.internmc.facebook.com/intern/diff/D31148926)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148926).

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 1]
```diff
@@ -1341,27 +1343,123 @@ Tensor computeConv2d(
...
-          inputs.size() == 2 || inputs.size() == 3,
-          buildErrorMessage("Invalid number of input operands"));
-      return (inputs.size() > 2)
-          ? computeTwoOperandWithAlpha(
-                "aten_add", inputs, outputShape, outputType, add_lambda)
-          : computeTwoOperand(
-                "aten_add", inputs, outputShape, outputType, add_lambda);
-    } break;
-    case aten::sub: {
+  auto A = c10::get<BufHandle>(inputs[0]);
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/kernel.cpp:1351]: I'm assuming this function and the ones below (`computeTranspose` and `computeExpand`) have just been refactored from below. So, not checking them in detail.

**Review by navahgar — APPROVED** [review]
LGTM

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148926).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148926).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148926).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148926).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148926).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148926).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148926).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148926).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148926).

**Pull request closed without merging**